src/Diagnostics.DataProviders/DataProviders/ChangeAnalysisDataProvider.cs

```diff
@@ -0,0 +1,173 @@
+            DateTime maxDataRetentionDate = DateTime.Now.AddDays(-14);
+            if (startTime < maxDataRetentionDate)
+            {
+                throw new Exception("Changes beyond last 14 days are not available. Please provide dates within last 14 days");
+            }
+
+            ChangeSetsRequest request = new ChangeSetsRequest
+            {
+                ResourceId = armResourceUri,
+                StartTime = startTime,
+                EndTime = endTime
+            };
+
+            // Get changeSet of the given arm resource uri
+            return await changeAnalysisClient.GetChangeSetsAsync(request);
+        }
+
+        /// <summary>
+        /// Get all Changes for a given changesetId.
+        /// </summary>
+        /// <param name="changeSetId">ChangeSetId retrieved from <see cref="GetChangeSetsForResource(string, DateTime, DateTime)"/>.</param>
+        /// <param name="resourceUri">ARM Resource Uri.</param>
+        /// <returns>List of changes.</returns>
+        public async Task<List<ResourceChangesResponseModel>> GetChangesByChangeSetId(string changeSetId, string resourceUri)
+        {
+            if (string.IsNullOrWhiteSpace(changeSetId))
+            {
+                throw new ArgumentNullException(nameof(changeSetId));
+            }
+
+            if (string.IsNullOrWhiteSpace(resourceUri))
+            {
+                throw new ArgumentNullException(nameof(resourceUri));
+            }
+
+            ChangeRequest request = new ChangeRequest
+            {
+                ResourceId = resourceUri,
+                ChangeSetId = changeSetId
+            };
+
+            return await changeAnalysisClient.GetChangesAsync(request);
+        }
+
+        /// <summary>
+        /// Gets ARM Resource Uri and Hostname for all dependent resources for a given site name.
+        /// </summary>
+        /// <param name="sitename">Name of the site.</param>
+        /// <param name="subscriptionId">Azure Subscription Id.</param>
+        /// <param name="stamp">Stamp where the site is hosted.</param>
+        /// <param name="startTime">End time of the time range.</param>
+        /// <param name="endTime">Start time of the time range.</param>
+        /// <returns>List of ARM resource uri for dependent resources of the site.</returns>
+        public async Task<ResourceIdResponseModel> GetDependentResourcesAsync(string sitename, string subscriptionId, string stamp, string startTime, string endTime)
+        {
+            // Query Kusto to get Dependencies
+            string query =
+                $@"
+                    let startTime = datetime({startTime});
+                    let endTime = datetime({endTime});
+                    let period = 30m;
+                    let sitename = ""{sitename}"";
+                    AntaresRuntimeWorkerSandboxEvents
+                    | where TIMESTAMP  < endTime and TIMESTAMP >= ago(20d) and Sandbox =~ sitename and EventId == 60011 
```

Comment:
Somewhere in the pipeline, do we ensure or throw the error that its not supported for Linux right now?

Reply:
We don't. I wasn't sure if we should really a throw an error because the query will run but won't give any results. So technically it doesnt "break" anything. I also thought when Linux team adds those tables in the future, we can tweak this method to check AppType. Let me know your thoughts.